### ISSUE TITLE
LT02: indent_unit = tab ignored for tsql dialect — always expects spaces

### ISSUE THREAD

**Issue by fank** [sqlfluff/sqlfluff#7545]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Setting `indent_unit = tab` in `.sqlfluff` has no effect when using the `tsql` dialect. LT02 still expects space-based indentation (e.g. "Expected indent of 4 spaces") even though the configuration explicitly requests tabs.

Every single indented line in every file triggers a false positive, making LT02 unusable for T-SQL projects that use tab indentation.

### Expected Behaviour

With `indent_unit = tab`, the following tab-indented query should pass LT02:

```sql
SELECT
	order_id
	,customer_name
FROM
	dbo.orders AS orders
WHERE
	orders.is_active = 1
```

### Observed Behaviour

```
L:   2 | P:   1 | LT02 | Expected indent of 4 spaces. [layout.indent]
L:   4 | P:   1 | LT02 | Expected indent of 4 spaces. [layout.indent]
L:   6 | P:   1 | LT02 | Expected indent of 4 spaces. [layout.indent]
```

Every tab-indented line is flagged. The `indent_unit = tab` setting is ignored entirely.

### How to reproduce

**.sqlfluff:**
```ini
[sqlfluff]
dialect = tsql
templater = raw
indent_unit = tab
```

**test.sql** (tab-indented):
```sql
SELECT
	order_id
FROM
	dbo.orders AS orders
```

```bash
sqlfluff lint --rules LT02 test.sql
```

Output:
```
== [test.sql] FAIL
L:   2 | P:   1 | LT02 | Expected indent of 4 spaces. [layout.indent]
L:   4 | P:   1 | LT02 | Expected indent of 4 spaces. [layout.indent]
```

**Note:** The same file passes LT02 when using `dialect = ansi`, so this appears to be specific to the tsql dialect overriding the tab configuration.

### Dialect

tsql

### Version

sqlfluff 4.0.4, Python 3.10.12

### Configuration

```ini
[sqlfluff]
dialect = tsql
templater = raw
indent_unit = tab
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by keraion** [comment]
Indentation configuration should be like so:
```ini
[sqlfluff]
dialect = tsql
templater = raw

[sqlfluff:indentation]
indent_unit = tab
```

**Comment by fank** [comment]
Thanks @keraion — that was indeed a misconfiguration on our side. Moving `indent_unit = tab` to `[sqlfluff:indentation]` resolved it. Closing.
